analyses/infercnv-consensus-cell-type/template-notebooks/ewings-infercnv-results.Rmd

```diff
@@ -0,0 +1,536 @@
+ref_dir <- file.path(analysis_dir, "references")
+infercnv_dir <- file.path(analysis_dir, "results", "SCPCP000015", params$sample_id)
+infercnv_immune_dir <- file.path(infercnv_dir, "ref-all-immune")
+infercnv_subset_dir <- file.path(infercnv_dir, "ref-subset-immune")
+
+# OpenScpCA data and results directories
+data_dir <- file.path(repository_base, "data", "current")
+```
+
+#### Input and output files
+
+```{r paths}
+# TSV with expected Ewing CNVs
+validation_tsv <- file.path(ref_dir, "cnv-validation.tsv")
+
+# References themselves so we know which cells belong to reference
+immune_ref_sce_file <- file.path(ref_dir, "normal-references", "SCPCP000015", "ref-all-immune.rds")
+subset_ref_sce_file <- file.path(ref_dir, "normal-references", "SCPCP000015", "ref-subset-immune.rds")
+
+# input infercnv files from both references, "all-immune" and "subset-immune"
+immune_hmm_tsv <- file.path(infercnv_immune_dir, glue::glue("{params$library_id}_cnv-metadata.tsv"))
+immune_png <- file.path(infercnv_immune_dir, glue::glue("{params$library_id}_infercnv.png"))
+subset_hmm_tsv <- file.path(infercnv_subset_dir, glue::glue("{params$library_id}_cnv-metadata.tsv"))
+subset_png <- file.path(infercnv_subset_dir, glue::glue("{params$library_id}_infercnv.png"))
+
+# ewings and consensus cell types
+ewings_tsv <- file.path(
+  data_dir,
+  "results",
+  "cell-type-ewings",
+  "SCPCP000015",
+  params$sample_id,
+  glue::glue("{params$library_id}_ewing-celltype-assignments.tsv")
+)
+
+# processed sce file
+sce_file <- file.path(
+  data_dir,
+  "SCPCP000015",
+  params$sample_id,
+  glue::glue("{params$library_id}_processed.rds")
+)
+```
+
+### Read in data
+
+```{r message = FALSE}
+validation_df <- readr::read_tsv(validation_tsv) |>
+  dplyr::filter(diagnosis == "Ewing sarcoma")
+
+immune_ref_barcodes <- readr::read_rds(immune_ref_sce_file) |>
+  colnames() |>
+  stringr::str_split_i("-", 2)
+subset_ref_barcodes <- readr::read_rds(subset_ref_sce_file) |>
+  colnames() |>
+  stringr::str_split_i("-", 2)
+
+# note these tsvs already have an indicator column for the reference
+# the `subcluster` contains the inferCNV-calculated clusters
+infercnv_df <- readr::read_tsv(immune_hmm_tsv) |>
+  dplyr::bind_rows(
+    readr::read_tsv(subset_hmm_tsv)
+  ) |>
+  tidyr::separate(cell_id, into = c("library_id", "barcodes"))
+
+# Read SCE and save UMAP coordinates and cell types
+umap_df <- readr::read_rds(sce_file) |>
+  scuttle::makePerCellDF(use.dimred = "UMAP") |>
+  dplyr::select(
+    barcodes,
+    UMAP1 = UMAP.1,
+    UMAP2 = UMAP.2
+  ) |>
+  # TODO: We currently are bringing in cell types from separate files, but in the next
+  # data release they will be in the SCE objects
+  dplyr::inner_join(
+    readr::read_tsv(ewings_tsv) |>
+      dplyr::select(barcodes, consensus_annotation, ewing_annotation)
+  )
+```
+
+
+## InferCNV Heatmaps
+
+Below, we show the `inferCNV` heatmaps from the `immune-all` and `immune-subset` normal references for this library.
```

Comment:
I actually think for this sample these heatmaps look pretty convincing in terms of seeing CNVs in the queried cells and not as much in the reference.